controller/volume_controller.go

```diff
@@ -2805,6 +2829,60 @@ func (c *VolumeController) updateRequestedBackupForVolumeRestore(v *longhorn.Vol
 	return nil
 }
 
+func (c *VolumeController) checkAndInitVolumeOfflineReplicaRebuilding(v *longhorn.Volume, rs map[string]*longhorn.Replica) error {
+	log := getLoggerForVolume(c.logger, v)
+
+	if v.Spec.BackendStoreDriver == longhorn.BackendStoreDriverTypeLonghorn {
+		return nil
+	}
+
+	if v.Status.OfflineReplicaRebuildingRequired {
+		return nil
+	}
```

Comment:
This looks weird... I know this field is determined by the healthy replica count **and** the below new config value. And this line is actually the result of the healthy replica count.
check... Can we/Do we need to move the check to this function instead?

Reply:
Is the update of field `v.Status.OfflineReplicaRebuildingRequired` affected by the previous value? I think the Longhorn will request auto attachment **only when** both conditions (healthy replica count < spec number, the config is enabled) are satisfied, no matter what the previous status is. 
For example, when the volume is already auto attached and the offline rebuilding is in progress, suddenly the user disables the setting, `v.Status.OfflineReplicaRebuildingRequired` should be false and the volume should be detached immediately. And the current rebuilding will be discarded.

Reply:
One question, should we set `v.Status.OfflineReplicaRebuildingRequired` to `false` in your case?

Reply:
YES for the example I mentioned. Any other concerns?

Reply:
No concern for now.